contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,350 @@
+            "Max validators reached"
+        );
+
+        require(
+            stakeETHTally + requiredETH <= stakeETHThreshold,
+            "Staking ETH over threshold"
+        );
+        stakeETHTally += requiredETH;
+
+        // Convert required ETH from WETH
+        IWETH9(WETH_TOKEN_ADDRESS).withdraw(requiredETH);
+        _wethWithdrawnAndStaked(requiredETH);
+
+        /* 0x01 to indicate that withdrawal credentials will contain an EOA address that the sweeping function
+         * can sweep funds to.
+         * bytes11(0) to fill up the required zeros
+         * remaining bytes20 are for the address
+         */
+        bytes memory withdrawal_credentials = abi.encodePacked(
+            bytes1(0x01),
+            bytes11(0),
+            address(this)
+        );
+
+        uint256 validatorsLength = validators.length;
+        // For each validator
+        for (uint256 i = 0; i < validatorsLength; ) {
+            bytes32 pubKeyHash = keccak256(validators[i].pubkey);
+            VALIDATOR_STATE currentState = validatorsStates[pubKeyHash];
+
+            require(
+                currentState == VALIDATOR_STATE.REGISTERED,
+                "Validator not registered"
+            );
+
+            IDepositContract(BEACON_CHAIN_DEPOSIT_CONTRACT).deposit{
+                value: 32 ether
+            }(
+                validators[i].pubkey,
+                withdrawal_credentials,
+                validators[i].signature,
+                validators[i].depositDataRoot
+            );
+
+            emit ETHStaked(
+                pubKeyHash,
+                validators[i].pubkey,
+                32 ether,
+                withdrawal_credentials
+            );
+
+            validatorsStates[pubKeyHash] = VALIDATOR_STATE.STAKED;
+
+            unchecked {
+                ++i;
+            }
+        }
+        // save gas by changing this storage variable only once rather each time in the loop.
+        activeDepositedValidators += validatorsLength;
+    }
+
+    // slither-disable-end reentrancy-eth
+
+    /// @notice Registers a new validator in the SSV Cluster.
+    /// Only the registrator can call this function.
+    /// @param publicKey The public key of the validator
+    /// @param operatorIds The operator IDs of the SSV Cluster
+    /// @param sharesData The validator shares data
+    /// @param ssvAmount The amount of SSV tokens to be deposited to the SSV cluster
+    /// @param cluster The SSV cluster details including the validator count and SSV balance
+    // slither-disable-start reentrancy-no-eth
+    function registerSsvValidator(
+        bytes calldata publicKey,
+        uint64[] calldata operatorIds,
+        bytes calldata sharesData,
+        uint256 ssvAmount,
+        Cluster calldata cluster
+    ) external onlyRegistrator whenNotPaused {
+        bytes32 pubKeyHash = keccak256(publicKey);
+        require(
+            validatorsStates[pubKeyHash] == VALIDATOR_STATE.NON_REGISTERED,
```

Comment:
The other methods do ` VALIDATOR_STATE currentState = validatorsStates[pubKeyHash];` then compare `currentState` in the require. Either way is fine, but let's be consistent. I lean towards the temp variable since we aren't running into stack space issues in these.

Reply:
I've added the temp variable `currentState`